File: FT_PC/FT_PC/Util/JsonUtil.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FT_PC
{
    class JsonUtil
    {
        /// <summary>
        /// 对将要输出的字符串进行编码
        /// </summary>
        /// <param name="str"></param>
        /// <returns></returns>
        public static String EncodeStr(params String[] str)
        {
            StringBuilder encodedStr = new StringBuilder();

            for (int i = 0; i < str.Length; i += 2)
            {
                encodedStr.Append(str[i]).Append("=").Append(str[i + 1]);
                if (i + 2 != str.Length)
                {
                    encodedStr.Append("&");
                }
            }

            return encodedStr.ToString();
        }

        /// <summary>
        /// 对收到的字符串进行解码
        /// </summary>
        /// <param name="str"></param>
        /// <returns></returns>
        public static Dictionary<String, String> parserStr(String str)
        {
            Dictionary<String, String> dic = new Dictionary<String, String>();

            //拆分字符串
            String[] sprStr = new String[10];
            sprStr = str.Split('&');
            foreach (String s in sprStr)
            {
                int i = s.IndexOf('=');
                String key = s.Substring(0, i);
                String value = s.Substring(i + 1, s.Length - i - 1);
                dic.Add(key, value);
            }

            return dic;
        }
    }
}

File: FT/app/src/main/java/com/jcx/pc/util/JsonUtil.java
package com.jcx.pc.util;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Cui on 16-6-8.
 */
public class JsonUtil {

    /**
     * 将数据以简直对的形式传递
     * @param str
     * @return
     */
    public static String EncodeStr(String... str){
        StringBuffer encodedStr=new StringBuffer();
        for (int i = 0; i < str.length; i+=2) {
            encodedStr.append(str[i]).append("=").append(str[i+1]);
            if (i+2 != str.length) {
                encodedStr.append("&");
            }
        }

        return encodedStr.toString();
    }

    public static Map<String,String> ParserStr(String str){

        if (str == null) {
            return null;
        }
        Map<String, String> maps = new HashMap<String, String>();
        //拆分字符串
        String[] sprStr=new String[10];
        sprStr=str.split("&");
        for (String string : sprStr) {
            int i=string.indexOf("=");
            String key=string.substring(0,i);
            String value=string.substring(i+1, (string.length()));
            maps.put(key, value);
        }

        return maps;
    }

}
